evalml/pipelines/time_series_regression_pipeline.py

```diff
@@ -205,39 +205,83 @@ def get_prediction_intervals(
         )
         if coverage is None:
             coverage = [0.95]
 
         if self.estimator.model_family in self.NO_PREDS_PI_ESTIMATORS and has_stl:
+            if self.problem_type == ProblemTypes.MULTISERIES_TIME_SERIES_REGRESSION:
+                from evalml.pipelines.utils import stack_data, unstack_multiseries
+
+                X, y = unstack_multiseries(
+                    X,
+                    y,
+                    self.series_id,
+                    self.time_index,
+                    self.input_target_name,
+                )
+
+            X_no_datetime, y_no_datetime = self._drop_time_index(X, y)
+
+            estimator_input = self.transform_all_but_final(
+                X_no_datetime,
+                y_no_datetime,
+                X_train=X_train,
+                y_train=y_train,
+            )
             pred_intervals = self.estimator.get_prediction_intervals(
                 X=estimator_input,
                 y=y,
                 coverage=coverage,
             )
-            trans_pred_intervals = {}
+            intervals_labels = list(list(pred_intervals.values())[0].keys())
+            interval_series_pred_intervals = {
+                interval: {} for interval in intervals_labels
+            }
             residuals = self.estimator.predict(
                 estimator_input,
-            )  # Get residual values
-            trend_pred_intervals = self.get_component(
-                "STL Decomposer",
-            ).get_trend_prediction_intervals(y, coverage=coverage)
-            for key, orig_pi_values in pred_intervals.items():
-                trans_pred_intervals[key] = pd.Series(
-                    (orig_pi_values.values - residuals.values)
-                    + trend_pred_intervals[key].values
-                    + y.values,
-                    index=orig_pi_values.index,
-                )
+            )
+            trans_pred_intervals = {}
+            if self.problem_type == ProblemTypes.MULTISERIES_TIME_SERIES_REGRESSION:
+                trend_pred_intervals = self.get_component(
+                    "STL Decomposer",
+                ).get_trend_prediction_intervals(y_no_datetime, coverage=coverage)
+                for series_id, intervals in pred_intervals.items():
+                    for key, orig_pi_values in intervals.items():
+                        series_id_target_name = (
+                            self.input_target_name + "_" + str(series_id)
+                        )
+                        interval_series_pred_intervals[key][
+                            series_id_target_name
+                        ] = pd.Series(
+                            (orig_pi_values.values - residuals[series_id].values)
+                            + trend_pred_intervals[series_id_target_name][key].values
+                            + y[series_id_target_name].values,
+                            index=orig_pi_values.index,
+                        )
```

Comment:
This is a lot of repeated code with the other logical branch, which is going to make life very hard for us if we ever need to update this code. Could you abstract it out into a local helper function?

Reply:
Something like 
```
def _get_series_intervals(intervals, residuals, trend_pred_intervals, y):
    return_intervals = {}
    for key, orig_pi_values in intervals.items():
        return_intervals[key] = pd.Series(
            (orig_pi_values.values - residuals.values)
            + trend_pred_intervals[key].values
            + y.values,
            index=origin_pi_values.index
        )
    return return_intervals

if is_multiseries(problem_type):
    for series_id, series_intervals in pred_intervals.items():
        series_id_target_name = self.input_target_name + "_" + str(series_id)
        interval_series_pred_intervals[series_id_target_name] = _get_series_intervals(
            series_intervals,
            residuals[series_id],
            trend_pred_intervals[series_id_target_name],
            y[series_id_target_name]
        )
else:
    trans_pred_intervals = _get_series_intervals(pred_intervals, residuals, trend_pred_intervals, y)
```

Reply:
The code I suggested does make a change to the dictionary structure for the multiseries case, which you'll have to let me know if it works or not - I swapped the intervals with series ids, to give us `{series_1: {0.75_lower: <>, 0.75_upper: <>, ...}, series_2: {...}...}` instead of `{0.75_lower: {series_1: <>, series_2: <>, ...}, ...}`
Personally, I think this would make it easier to get per-series prediction intervals, but you'll have to let me know if it's too much effort to swap things around at this point. We could also completely overhaul the data structure for this to be something actually 2D like a dataframe instead of nested dictionaries, but that might just be tech debt for the future.